### ISSUE TITLE
Optimization issue with CurrentMediaSessions

### ISSUE THREAD

**Issue by DubyaDude** [DubyaDude/WindowsMediaController#13]
Currently, grabbing `CurrentMediaSessions` creates a new `ReadOnlyDictionary`.
This can cause massive memory/general optimization issues for anything that calls `CurrentMediaSessions`.

Need to create a class to handle both an internally usable `Dictionary`, and an externally usable `ReadOnlyDictionary`. Keeping 2 dictionaries in sync **should** be much more efficient than creating a new dictionary every `CurrentMediaSessions` get.

**Comment by VolcanicArts** [comment]
```c#
public IReadOnlyDictionary<string, MediaManager.MediaSession> CurrentMediaSessions => _CurrentMediaSessions;
private readonly Dictionary<string, MediaManager.MediaSession> _CurrentMediaSessions = new Dictionary<string, MediaManager.MediaSession>();
```

Pretty sure changing to this will work?

**Comment by DubyaDude** [comment]
Yup! That worked wonderfully! Thanks a ton.

**Comment by DubyaDude** [comment]
Implemented in 2.4.0 release
